coderd/portsharing/portsharing.go

```diff
@@ -0,0 +1,13 @@
+package portsharing
+
+type PortSharer interface {
+	CanRestrictSharing() bool
```

Comment:
I think this interface would be better as:
- `CanRestrictSharing() bool`
- `SetTemplateSharingRestriction(ctx context.Context, db database.Store, template database.Template, level database.PortShareLevel) (database.Template, error)`
    - Which would do nothing in AGPL
- `CreatePortShare(ctx context.Context, ...) (database.PortShare, error)`
    - Which would enforce the template's restriction in enterprise, or not enforce anything in AGPL

Reply:
I updated the interface to carry more logic, but not exactly the proposition you chose since it actually resulted in some duplicate code between the two implementations.